linkerd/app/outbound/src/endpoint.rs

```diff
@@ -209,10 +209,14 @@ impl<S> Outbound<S> {
     pub fn push_endpoint<I>(self) -> Outbound<svc::ArcNewTcp<tcp::Endpoint, I>>
     where
         Self: Clone + 'static,
-        S: svc::Service<tcp::Connect, Error = io::Error> + Clone + Send + Sync + Unpin + 'static,
-        S::Response:
-            tls::HasNegotiatedProtocol + io::AsyncRead + io::AsyncWrite + Send + Unpin + 'static,
-        S::Future: Send + Unpin,
+        S: svc::MakeConnection<tcp::Connect, Metadata = Local<ClientAddr>, Error = io::Error>
+            + Clone
+            + Send
+            + Sync
+            + Unpin
+            + 'static,
```

Comment:
style nit, doesn't actually matter: consider breaking this into two lines?
```suggestion
        S: svc::MakeConnection<tcp::Connect, Metadata = Local<ClientAddr>, Error = io::Error>
        S: Clone + Send + Sync + Unpin + 'static,
```